packages/api/src/controllers/savedSearch.ts

```diff
@@ -1,14 +1,28 @@
 import { z } from 'zod';
 
 import { SavedSearchSchema } from '@/common/commonTypes';
+import Alert from '@/models/alert';
 import { SavedSearch } from '@/models/savedSearch';
 
 type SavedSearchWithoutId = Omit<z.infer<typeof SavedSearchSchema>, 'id'>;
 
-export function getSavedSearches(teamId: string) {
-  return SavedSearch.find({
+export async function getSavedSearches(teamId: string) {
+  const savedSearches = await SavedSearch.find({
     team: teamId,
   });
+  const alerts = await Promise.all(
+    savedSearches.map(({ _id }) =>
+      Alert.find({ savedSearch: _id }, { __v: 0 }),
+    ),
+  );
```

Comment:
perf: we don't need to fire multiple requests to get savedSearch alerts here. Instead, we can query all savedSearch alerts given the team id:
```
const alerts = Alert.find({
  team: teamId,
  savedSearch: { $exists: true, $ne: null }
}, { __v: 0, ONLY SELECT THINGS USED BY THE APP})
```
We can run this alerts query and the savedSearches one concurrently.
Then, we can either send an alerts lookup table or join them with saverSearches back

Reply:
done